File: app/src/main/java/com/mlkit/textrecognition/presentation/utils/SchedulersFacade.java
package com.mlkit.textrecognition.presentation.utils;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class SchedulersFacade {

    @Inject
    public SchedulersFacade() {
    }

    public Scheduler io() {
        return Schedulers.io();
    }

    public Scheduler ui() {
        return AndroidSchedulers.mainThread();
    }

    public Scheduler computation() {
        return Schedulers.computation();
    }

    public Scheduler single() {
        return Schedulers.single();
    }
}

File: app/src/main/java/com/mlkit/textrecognition/presentation/utils/SchedulersFacade.kt
package com.mlkit.textrecognition.presentation.utils

import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

object SchedulersFacade {

    fun io(): Scheduler {
        return Schedulers.io()
    }

    fun ui(): Scheduler {
        return AndroidSchedulers.mainThread()
    }

    fun computation(): Scheduler {
        return Schedulers.computation()
    }

    fun single(): Scheduler {
        return Schedulers.single()
    }
}